crates/viewer/re_space_view_spatial/src/mesh_loader.rs

```diff
@@ -193,6 +218,37 @@ impl LoadedMesh {
     }
 }
 
+fn add_albedo_factor_to_mesh(
+    mesh_instances: Vec<MeshInstance>,
+    asset3d: &Asset3D,
+    render_ctx: &RenderContext,
+) -> Result<Vec<MeshInstance>, MeshError> {
+    if let Some(m) = &mesh_instances[0].mesh {
+        // Create new material from mesh_instance, add Asset3D's albedo_factor
+        let material = mesh::Material {
+            albedo_factor: asset3d
+                .albedo_factor
+                .map_or(re_renderer::Rgba::WHITE, |c| c.0.into()),
+            ..m.materials[0].clone()
+        };
+
+        let mesh_inst = Arc::clone(m);
+
+        // Create new mesh with albedo_factor
+        let mesh = mesh::Mesh {
+            materials: smallvec![material],
+            ..(*mesh_inst).clone()
+        };
+
+        Ok(vec![MeshInstance::new_with_cpu_mesh(
+            Arc::new(GpuMesh::new(render_ctx, &mesh)?),
+            Some(Arc::new(mesh)),
+        )])
+    } else {
+        Ok(mesh_instances)
+    }
+}
```

Comment:
it looks like you're jumping through many many hoops & clones instead of just making `MeshInstance` `mut` and swap out the albedo factor. What you want to do is:
* if no albedo factor on mesh defined, early out
* otherwise, go through all meshes, edit the albedo factor

Reply:
Correct me if I'm wrong, but the `mesh` of `MeshInstance` is an `Option<Arc<Mesh>>`, so I can't modify it because it was instantiated without Mutex.
I tried using `Arc::get_mut()`, it return Some(Mesh) so there is no other Arc.
It works : back in `load_asset3d_parts` after the call of `add_albedo_factor_to_mesh`, `albedo_factor` in `MeshInstance.mesh` is modified. But for some reason the asset still appears white.
Do you have an idea of what is happening here ? Could it be overwritten somewhere else ?

```
fn add_albedo_factor_to_mesh(
    mesh_instances: &mut Vec<MeshInstance>,
    asset3d: &Asset3D,
) {
  for instance in mesh_instances {
      if let Some(mesh) = &mut instance.mesh {
          if let Some(mutable_mesh) = Arc::get_mut(mesh) {
              mutable_mesh.materials[0].albedo_factor = asset3d
                  .albedo_factor
                  .map_or(re_renderer::Rgba::WHITE, |c| c.0.into());
          } else {
              println!("Arc is none");
          }
      }
  }
}
```

Reply:
oh you're right, totally missed that this is a change on the mesh rather the mesh instances. This makes things actually more complicated still. What you're doing right now creates a lot more new meshes potentially: in a gltf a single mesh may be instantiated many times, which is why its an arc in the first place. We really don't want to duplicate meshes since this may copy around a looot of memory.
... but the vec of mesh instances itself can still be edited in-place ;-)

I can take a look how to solve this best

Reply:
it would be ofc a lot easier to just change the `additive_tint` property, but that's confusing because we're not even exposing this in the meshes archetype. (it's a remnant from back when we didn't have outlines and needed some way of highlighting things)

Reply:
The reason you don't see any changes when changing the color oh the _optional_ cpu mesh is that it does not affect the gpumesh. Creating that is also quite expensive since it entails transfer of all the gpu data to cpu.
So what we really have to do here is to change the loaders to not do that step. Then it's much easier to do postprocessing of the resulting cpu-only meshes